src/Application/Messenger/Handler/StripePayoutHandler.php

```diff
@@ -153,8 +159,11 @@ public function __invoke(StripePayout $payout): void
             }
         }
 
-        if ($count > 0) {
+        // Ensure we always close a transaction if we opened one, even if no charge IDs
+        // were applicable to update.
+        if (count($chargeIds) > 0) {
             $this->entityManager->flush();
+            $this->entityManager->commit();
         }
```

Comment:
@NoelLH what do you think about using `EntityManagerInterface::wrapInTransaction` instead of `beginTransaction`? Would mean we wouldn't have to remember to put the `commit` in. Or we could use transactional, which we have our implementation of in `\MatchBot\Application\Persistence\RetrySafeEntityManager::transactional`, but it is for some reason deprecated by Doctrine.

![image](https://user-images.githubusercontent.com/159481/213735163-5f46abf2-ba2d-4369-bf78-24a9048c6754.png)

(no need for flush, wrapInTransaction does that too)

Reply:
@bdsl I think the last time I looked at it in the context of MatchBot, it was going to make certain use cases more complex because of the nesting and number of rollback points, make unit testing harder, or both. I can't remember exactly why it wasn't tenable though – maybe here it will become an option if we move to smaller txns